### ISSUE TITLE
P1 experimental symbolic analysis: admit AnalysisGNN runtime on trusted scores

### ISSUE THREAD

**Issue by gr-rr** [ListenCloser/listencloser#1248]
Parent symbolic depth: #1178. Theory truth owner: #1020. Runtime/product landscape: #1246. Representation authority: #613. Source-score authority: #1082.

## Outcome

Make official **AnalysisGNN** pretrained inference reproducibly runnable on one exact trusted MusicXML/score Version and normalize a bounded subset of its predictions into method-qualified experimental symbolic evidence.

This is a **runtime-admission slice**, not a benchmark tournament and not a replacement for current theory authority.

## Why now

#1246 identifies AnalysisGNN as unusually high leverage because one score-native engine spans multiple useful symbolic tasks (e.g. key/RN/cadence/phrase/section and related labels). The immediate question is operational: can its actual public pretrained path be pinned and run cleanly against ListenCloser inputs without contaminating the normal worker?

## First slice

1. Pin the official AnalysisGNN source revision/package and exact pretrained artifact identity.
2. Pin W&B/pretrained artifact acquisition deterministically; avoid mutable `latest` references.
3. Record repository license separately from model/artifact/training-data terms.
4. Prefer an isolated runtime if `pyg-lib`, GraphMuse, Torch/PyG, W&B or Python constraints would perturb the main worker.
5. Run one repository-lawful trusted MusicXML fixture through official inference.
6. Start with a deliberately small normalized output set: local/global key where actually emitted, Roman-numeral/chord labels, cadence and phrase/section proposals. Do not expose every upstream task merely because it exists.
7. Preserve exact source-score Version, upstream task/model identity, preprocessing and method-specific scores/probabilities without relabeling them as generic confidence.
8. Treat predictions as experimental proposals. Current music21/TheoryInterpreter truth and #1020 withholding rules remain authoritative until separately changed.
9. Explicit failure only; no silent substitution with current theory rules.
10. Record cold load, warm inference, peak RSS/VRAM and package footprint.

## Boundaries

- trusted source score/MusicXML only for the first slice;
- no inference of Score↔performance alignment;
- no new symbolic dashboard;
- no generic graph/relation framework;
- no new handcrafted cadence/local-key rules;
- no broad corpus benchmark before runtime admission;
- no promotion of AnalysisGNN predictions to production theory truth in this issue.

## Runtime admission proof

```text
upstream repo + exact revision
package/runtime versions
code license
model/artifact license + training-data restrictions
exact artifact identity/hash where practical
runtime boundary
reproducible artifact download/cache path
one real trusted-score inference
bounded normalized LC output
exact source Version lineage
cold/warm runtime
peak RSS/VRAM
explicit failure behavior
capability classification
```

## Acceptance

- [ ] official pretrained AnalysisGNN inference runs reproducibly from pinned assets;
- [ ] one exact trusted MusicXML Version produces non-empty bounded symbolic predictions;
- [ ] exact source/model/task/preprocessing provenance persists;
- [ ] dependency isolation is explicit and does not destabilize the normal worker;
- [ ] current theory authority is not silently replaced;
- [ ] no silent fallback exists;
- [ ] runtime/resource observations are recorded;
- [ ] focused backend tests and canonical gates are green;
- [ ] no broad evaluator or UI redesign is introduced.

Refs #1178 #1020 #1246 #613 #1082

**Comment by gr-rr** [comment]
First operational pass is now on `agent/1248-analysisgnn-runtime`.

Pinned upstream revision: `e115182fb29b74bdcb6bf3547ed427d967580947` (package declares `1.0.0`, MIT code). The useful production-shaped route is to **avoid upstream's implicit W&B downloader entirely**: official prediction accepts both `--checkpoint_path` and `--input_score`, so the LC adapter requires a pre-provisioned local checkpoint + SHA-256 and runs W&B offline.

The upstream default model identifier is `melkisedeath/AnalysisGNN/model-uvj2ddun:v1`, but the repository does not state the pretrained artifact's redistribution/commercial license. Until those artifact terms are independently pinned, the adapter marks the runtime `INTERNAL_ONLY` / model license `UNVERIFIED` and is not registered as product theory authority.

The branch currently adds the isolated local-checkpoint runner and focused fail-closed tests; next admission tail is acquiring the exact pretrained checkpoint legally, hashing it, running one trusted MusicXML fixture, and recording resource behavior.

**Comment by gr-rr** [comment]
Piano-first product direction update (2026-09-06): this is now the highest-priority missing analysis path. Lyrics/text work is deliberately deferred; do not broaden this slice beyond trusted symbolic piano/score analysis.

Started implementation in #1284. That PR adds the missing bounded normalization boundary for `cadence`, `localkey`, and `romanNumeral` with exact score locators while preserving `INTERNAL_ONLY` provenance. It intentionally does not make the model user-reachable or change theory authority.

After #1284, continue #1248 with the real admission tail in this order:
1. resolve exact default W&B checkpoint terms (repo code is MIT; model artifact remains separately unverified);
2. pin checkpoint SHA-256 and provision the isolated runtime;
3. run one lawful piano MusicXML fixture through the real checkpoint and record cold/warm latency + RSS/VRAM;
4. inspect actual output semantics/failure tails;
5. persist only the bounded normalized score evidence that survives that proof;
6. then open the musician-facing `Score Analysis · Experimental` surface in Score/Breakdown, with current LC theory remaining the default authority and overlapping RN/harmony exposed only as an alternate interpretation.

Do not treat `INTERNAL_ONLY` as the desired permanent product posture if the artifact/runtime proof succeeds.
